nevergrad/parametrization/data.py

```diff
@@ -0,0 +1,353 @@
+# LICENSE file in the root directory of this source tree.
+
+import warnings
+import typing as t
+import numpy as np
+from nevergrad.common.typetools import ArrayLike
+from . import core
+from ..instrumentation import transforms as trans  # TODO move along
+# pylint: disable=no-value-for-parameter
+
+
+BoundValue = t.Optional[t.Union[float, int, np.int, np.float, np.ndarray]]
+A = t.TypeVar("A", bound="Array")
+
+
+class BoundChecker:
+    """Simple object for checking whether an array lies
+    between provided bounds.
+
+    Parameter
+    ---------
+    a_min: float or None
+        minimum value
+    a_max: float or None
+        maximum value
+
+    Note
+    -----
+    Not all bounds are necessary (data can be partially bounded, or not at all actually)
+    """
+
+    def __init__(self, a_min: BoundValue = None, a_max: BoundValue = None) -> None:
+        self.bounds = (a_min, a_max)
+
+    def __call__(self, value: np.ndarray) -> bool:
+        """Checks whether the array lies within the bounds
+
+        Parameter
+        ---------
+        value: np.ndarray
+            array to check
+
+        Returns
+        -------
+        bool
+            True iff the array lies within the bounds
+        """
+        for k, bound in enumerate(self.bounds):
+            if bound is not None:
+                if np.any((value > bound) if k else (value < bound)):
+                    return False
+        return True
+
+
+# pylint: disable=too-many-arguments
+class Array(core.Parameter):
+    """Array variable of a given shape.
+
+    Parameters
+    ----------
+    init: np.ndarray, or None
+        initial value of the array (defaults to 0, with a provided shape)
+    shape: tuple of ints, or None
+        shape of the array, to be provided iff init is not provided
+    mutable_sigma: bool
+        whether the mutation standard deviation must mutate as well (for mutation based algorithms)
+
+    Note
+    ----
+    More specific behaviors can be obtained throught the following methods:
+    set_bounds, set_mutation, set_integer_casting
+    """
+
+    def __init__(
+            self,
+            *,
+            init: t.Optional[ArrayLike] = None,
+            shape: t.Optional[t.Tuple[int, ...]] = None,
+            mutable_sigma: bool = False
+    ) -> None:
+        sigma = Log(init=1.0, exponent=1.2, mutable_sigma=False) if mutable_sigma else 1.0
+        super().__init__(sigma=sigma, recombination="average")
+        err_msg = 'Exactly one of "init" or "shape" must be provided'
+        if init is not None:
+            if shape is not None:
+                raise ValueError(err_msg)
+            self._value = np.array(init, copy=False)
+        elif shape is not None:
+            assert isinstance(shape, tuple) and all(isinstance(n, int) for n in shape)
+            self._value = np.zeros(shape)
+        else:
+            raise ValueError(err_msg)
+        self.integer = False
+        self.exponent: t.Optional[float] = None
```

Comment:
Maybe it's easier to set a fixed exponent to say `e` and then change the sigma/initial value?
`(exponent=2 * e, sigma=1, init=(2 * e)^1)` should be equivalent to `(exponent=e, sigma=1 + ln(2), init=e^(1 + ln(2)))` no?
```
(2*e)^(x) -> (2*e)^(x + sigma) = (2 * e) ^ (x + 1)
e^(x.(1 + ln(2))) -> e^(x.(1 + ln(2)) + sigma) = e^(x.(1 + ln(2)) + (1 + ln(2))) = e^((x+1).(1 + ln(2))) = (2 * e) ^ (x + 1)
```

Reply:
I wanted to avoid the equations altogether :D 
I like it the current because that one can define mutations changes in one space or another
I don't like it the current way because both ways are equivalent so it's redundant...
For now I'm thinking less equations will hopefully make it less buggy, but I need to think about it today then.